src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1192,6 +1197,17 @@ private IEnumerable<TypeInferenceInfo> InferTypeInPropertyDeclaration(PropertyDe
             private IEnumerable<TypeInferenceInfo> InferTypeInBaseMethodDeclaration(BaseMethodDeclarationSyntax declaration)
             {
                 var methodSymbol = SemanticModel.GetDeclaredSymbol(declaration);
+                return GetReturnTypeFromMethodSymbol(methodSymbol);
+            }
+
+            private IEnumerable<TypeInferenceInfo> InferTypeInAccessorDeclaration(AccessorDeclarationSyntax accessorDeclaration)
```

Comment:
My idea was actually to replace both `InferTypeInBaseMethodDeclaration` and `InferTypeInAccessorDeclaration` with one method taking a `SyntaxNode`. I don't know what it would be called though. InferTypeInMethodLikeDeclaration? MethodSymbolDeclaration?

Reply:
Actually, the best approach might be to call GetDeclaredSymbol regardless of what we found (eliminating the switch statement) and then check if the symbol is a method symbol or a property symbol and get its type.

Reply:
I tried that, but I can't get it more readable / with less code that way. Getting the method symbol requires either the `BaseMethodDeclarationSyntax` or `AccessorDeclarationSyntax` to be passed to GetDeclaredSymbol. They don't share a base class. If you have a good idea, please let me know.

Reply:
ok, I thought there was an overload that takes a SyntaxNode

Reply:
There is, and it returns an `ISymbol`. But I don't know of a way to get the (necessary) method symbol based on that alone… If someone does know, I'd love to hear it.

Reply:
You'd have to check whether the symbol is an IMethodSymbol or an IPropertySymbol and then get its type.

Reply:
Actually, you could just use this extension method:
https://github.com/dotnet/roslyn/blob/84e1af7365956a04027014ce48c228f874a93d3c/src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs#L843

Reply:
Wait, no... it doesn't handle methods.

Reply:
Of course, pattern matching to the rescue again! See my newest commit, hope this is to your liking.